### ISSUE TITLE
External Signals For Buys & Sells

### ISSUE THREAD

**Issue by ManuGithu** [taniman/profit-trailer-enhancements#14]
Signalers monitor crypto currency and can send buy signals. Those signals are based on complex mathematical technical analysis and are processed externally.
 
There would no need for the user to configure a strategy with indicators such as EMAGAIN, LOWBB. The strategy would be to use an external signal that triggers buys.

**Comment by nicnic22** [comment]
This would be an awesome addition! +1

**Comment by ninadpachpute** [comment]
One way to implement is via telegram bot.

**Comment by iAmVaeVictus** [comment]
Since no one has mentioned it, TradingView signaling. Though, the only way to do it (legally) is via email triggering, which SUCKS. There is a group that claims to have found an API to do it, but that is suspect (seems that data scraping is the only other way to get it done RN)

**Comment by ManuGithu** [comment]
Example of services that send signals:

http://www.mininghamster.com
http://www.coindicator.io

**Comment by AkliOnet** [comment]
https://www.tradingview.com/symbols/BTCUSD/technicals/
I don't know if they have an api but trading view moving average resume is pretty accurate when the short and long mobing average give the same result. But if elroy introduce the possibility to use multiple indicator there is no need for that.

Their oscilators never agree ! even when it's obvious

**Comment by nric** [comment]
that would really value up the profit-trailer big time for me. 

An URL parameter to send the buy/sell signal for a pair would be an option to implement it.

**Comment by Slumberland** [comment]
For me this is the #1 missing feature.

I've even been considering just manually opening the signal positions from Mining Hamster for the time being. CH is an alternative but PT is further down the road.

**Comment by tactikill** [comment]
Isn't external signals using multiple of these buy strategies (EMACROSS, EMASPREAD, LOWBB, etc.) to tell you if it is a 'Buy" ? Say it hits on 3 out of your 4 settings, then it is a buy. The only reason you would need to purchase external triggers is to have someone set their "magic" formula up. Right? I do not own any external signalers. Personally, I would much rather have the ability to set up my own INTERNAL signalers before adding external. External is great, but inside first. Please.

**Comment by nicnic22** [comment]
I'm pretty sure it would be a lot more work to implement internal signals, for both them and you. Internal signals shouldnt be too hard to implement i imagine

**Comment by nric** [comment]
@tactikill , it is not that simple. All strategies you mentioned are focused on the same strategy: Find the dip and buy at the dip and then hope for a rise. It is hence not astounding that the results will largely be the same. That defines you win-chance, all other tools like DCA don’t change that probability (possibly the opposite, due to the large fluctuations in crypto I think DCA is quite dangerous to use). 

Yes, you can combine your mentioned regression-to-mean-methods (EMA,BB) which might bring some better specificity but less buys (equivalent to less risk = less opportunity, as always). Yes, in that context also profit-feeder can somewhat help, mainly by avoiding bad market false-positives from the methods you names. However, what you want in game theory is change your fundamental chances of winning.
In quant, everyone is focusing on more advanced tools, mainly machine learning with lately a heavy emphasis on deep learning via neural networks. In addition to the simple mean-curves methods or first derivatives, many many more parameters can be respected, learned and correlations found. Common parameters involve complex things like monitoring search words, website traffics, high-frequency sentiment analysis of news outlets, websites, forums and channels. Signals from automats like this have a way better chance of winning the game. The efficient market theory in it’s weakest form is simply not all-true anymore. 

Why still use profit trailer when using external deep NN for predictions? A) Because PT is relatively stable, good price and nice UI and especially B) PT has a great sell-trailing algorithm. From what I see it is currently still smarter to use trailing sell instead of prediction sell from depp NNs, however deep NNs are better for buy-decision making then any time mean method and first-derivative method can ever be (because it uses all of them and hundreds of others).  In other words it is hard (impossible) to predict how high something will go, is much easier to make data-based decisions IF something will now or soon rise.

In short: A interface to an external machine (or human+machine) making smart decision and letting the profit trailer just trail the profit would be tremendously useful in my humble opinion.

**Comment by tactikill** [comment]
Thank you @nric for the in depth explanation!

**Comment by kerrychoe** [comment]
+1

**Comment by nstefi** [comment]
This is exactly what I was looking for, and I think it would be the best feature to find better buy opportunities. I think the easiest and most flexible solution would be to work based on Tradingview alerts.
There is a bot that already uses that technique, it's called Autoview. I think it's a Chrome browser plugin, and it can detect the popup alerts from Tradingview. This is their website:
[https://autoview.with.pink/](https://autoview.with.pink/)
The advantage of using Tradingview is, that you could use PineScript with any complicated strategies combining multiple indicators. There are lots of scripts available online, or you make your own indicators or combine them and create your alerts. I thought about comining the 2 bots if this feature is not available, so I would use the Autoview for buys, and let Profit Trailer handle the sells, because trailing and watching for a set profit level is still a better strategy for selling than using indicators and alerts.

**Comment by Jackpot64** [comment]
+1

**Comment by truecaller96** [comment]
+1 This would add so much value to the great service we are already receiving

**Comment by nstefi** [comment]
I see you guys typing +1 in the message, but did you click on thumbs up on the original message?

**Comment by LeaderOfChampions** [comment]
So... all our bots get the same signal from an external source? Meaning a signal controls over 7000 bots to perform an action? Why wouldn't you want the efforts to improve the current core where it looks at strategies and enhances that first?

**Comment by nstefi** [comment]
No, not 1 single signal. I could have my own script in TradingView and define my own alerts, that would be my signal. Other people can use their own signal, and they may be trading other altcoins. Or use some telegram message as a signal.
Because implementing this to allow an external signal sounds much easier than implementing multiple indicators, and even if you allow to combine 2 indicators, that is still limited to what you can do with an external signal that could come from TradingView based on a custom script that can include tons of indicators and some special algorithms, etc. Or you could plug in some other bot that is specialized to generate signals with better buy opportunities. So the sky is the limit if you allow some external signal, and implementing it should be very easy, just tell the bot to buy a certain amount when it receives the signal.

**Comment by aycanurun** [comment]
+1

**Comment by elementalwindx** [comment]
Next thing you know is you get a signal that sells all your alts and buys all in on a shitcoins yeahhhhh

**Comment by Jaguar143** [comment]
This is cool feature to be added on PT where you can set Pairs you want to be signaled with buy - then signaled again when sell. Buy LOW sell HIGH notification. So +1 for me.

**Comment by tooraj-enayati** [comment]
Mining Hamster is the way to go!
https://www.mininghamster.com/bots/

**Comment by clownfish44** [comment]
https://github.com/taniman/profit-trailer/issues/272 already on 2.2
